academy/4-my-own-chain/ignitecli.md

```diff
@@ -63,7 +67,7 @@ This entire exercise was built using the Ignite CLI version noted above. Using a
 To install this specific version of Ignite CLI, use:
 
 ```sh
-curl https://get.ignite.com/cli@v0.17.0! | bash
+$ curl https://get.ignite.com/cli@v0.17.0! | bash
```

Comment:
Following up on that, here is a small list of changes between the versions:

```m̀d
# What changes compared to tutorial

List of the changes when using `v0.20.1` ipv `v0.17.0`

## Ignite CLI

### Your GUI

    - Replace `npm run serve` by `npm run dev`
    - Replace `localhost:8080` by `localhost:3000`

### Your first message

List of files modified:

```
modify proto/checkers/tx.proto
modify x/checkers/client/cli/tx.go
create x/checkers/client/cli/tx_create_post.go
modify x/checkers/handler.go
create x/checkers/keeper/msg_server_create_post.go
modify x/checkers/module_simulation.go
create x/checkers/simulation/create_post.go
modify x/checkers/types/codec.go
create x/checkers/types/message_create_post.go
create x/checkers/types/message_create_post_test.go
```

## Store Object - Make a Checkers Blockchain

### The stored game object

After scaffolding, this is missing from `x/checkers/types/keys.go`

```go
const (
    StoredGameKey = "StoredGame-value-"
)
```

### Protobuf objects

**`proto/checkers/next_game.proto`**:

The `NextGame` message does not contains a `creator` field:

```protobuf
message NextGame {
  uint64 idValue = 1; 
  
}
```

Someone already found this issue and suggested, for the latest cli version, to use these commands:

<https://github.com/ignite-hq/cli/issues/2234#issuecomment-1083250874>

This changes the order of the fields however.

**`proto/checkers/genesis.proto`**:

Extra field params in the message

```protobuf
message GenesisState {
  Params params = 1 [(gogoproto.nullable) = false];
  NextGame nextGame = 2;
  repeated StoredGame storedGameList = 3 [(gogoproto.nullable) = false];
  // this line is used by starport scaffolding # genesis/proto/state
}
```

Consequently, the same apply for the generated go struct.

### Files to adjust

Because of the order of the fields differs, initalizing the struct must be done with `&NextGame{uint64(0), ""}`.
Or maybe directly with the field names so that the order does not matter: `&NextGame{Creator: "", IdValue: uint64(0)}`

```